packages/openapi2typescript-cli/src/core/ApiDescriptorBuilder.ts

```diff
@@ -440,6 +459,76 @@ class ApiDescriptorBuilder {
 
         return schemas;
     }
+
+    private getTopParameters(): StringMap<ParameterWithTypeName> {
+        const parameters: StringMap<ParameterWithTypeName> = {};
+        if (this.openapi.components?.parameters) {
+            for (const [ typeName, parameter ] of sortByKey(Object.entries(this.openapi.components.parameters))) {
+                if (isReference(parameter)) {
+                    throw new Error('Invalid reference found at parameters level');
+                }
+
+                let schema: SchemaOrType;
+                if (parameter.schema) {
+                    schema = this.getSchema(parameter.schema);
+                } else {
+                    schema = {
+                        type: SchemaType.UNKNOWN
+                    };
+                }
+
+                const paramType = this.getParamType(parameter.in);
+
+                if (parameter.required || paramType === ParamType.PATH) {
+                    schema.isOptional = false;
+                } else if (!parameter.required) {
+                    schema.isOptional = true;
+                }
+
+                parameters[typeName] = {
+                    typeName,
+                    name: parameter.name,
+                    type: paramType,
+                    schema
+                } as ParameterWithTypeName;
+            }
+        }
+
+        return parameters;
+    }
+
+    private getTopRequestBodies(): StringMap<RequestBodyWithTypeName> {
+        const requestBodies: StringMap<RequestBodyWithTypeName> = {};
+        if (this.openapi.components?.requestBodies) {
+            for (const [ typeName, requestBody ] of sortByKey(Object.entries(this.openapi.components.requestBodies))) {
+                if (isReference(requestBody)) {
+                    throw new Error('Invalid reference found at request bodies level');
+                }
+
+                requestBodies[typeName] = {
+                    ...this.getRequestBody(requestBody),
+                    typeName
+                };
+            }
+        }
+
+        return requestBodies;
+    }
+
+    private getParamType(rawParamType: OpenAPI3.Parameter['in']): ParamType {
+        switch (rawParamType) {
+            case 'header':
+                return ParamType.HEADER;
+            case 'query':
+                return ParamType.QUERY;
+            case 'cookie':
+                return ParamType.COOKIE;
+            case 'path':
+                return ParamType.PATH;
+        }
+
+        assertNever(rawParamType);
```

Comment:
Does it make more sense to put the assertNever as the default case in the switch?

Reply:
I think it would make more sense to put in the default.
Functionally it's the same, but might be easier to read next to the default.